### PR TITLE
refactor(aws): consolidate AWS SDK v2 client construction onto AwsClientSupport

### LINKED ISSUE #7083: Migrate remaining AWS connectors to use `AwsClientSupport` from `aws-base`
__What should we do?__

`AwsClientSupport` was extracted in #7075 to deduplicate AWS SDK v2 client construction across the 4 AgentCore connectors. The same pattern (credential resolution → region → optional endpoint override → build) is used in other AWS connectors as well.

**Scope:** Audit the remaining AWS connectors (`aws-lambda`, `aws-sqs`, `aws-sns`, `aws-dynamodb`, `aws-eventbridge`, `aws-bedrock`, `aws-bedrock-knowledgebase`, `aws-s3`) and migrate them to `AwsClientSupport.createClient()` / `configureClient()` where applicable.

**Note:** Some connectors may still use AWS SDK v1, so this may not apply everywhere.

__Why should we do it?__

Reduces duplication of AWS SDK v2 client construction logic across AWS connectors, improves consistency, and makes future SDK upgrades / configuration changes easier to roll out in a single place.

Follow-up to discussion in https://github.com/camunda/connectors/pull/7075#discussion_r3160995937.

### PR DESCRIPTION
**Pull request by chillleader**
## Description

Issue #7083 asked to consolidate the AWS SDK v2 client-construction code scattered across the AWS connectors onto the shared `AwsClientSupport` helper (already used by aws-dynamodb, aws-textract, aws-comprehend and aws-bedrock-knowledgebase). Each connector was building its client by hand: resolving credentials via `CredentialsProviderSupportV2`, reading `configuration.region()`/`configuration.endpoint()`, and wiring them onto the SDK builder itself, with small variations per connector. That logic is now centralized in one place.

This PR migrates the remaining 7 AWS connectors:

- aws-lambda
- aws-sqs (outbound and inbound)
- aws-sns
- aws-eventbridge
- aws-bedrock
- aws-s3
- aws-sagemaker

For connectors with a deprecated per-resource region field (lambda's `awsFunction.region`, sns's per-topic region, sqs's per-queue region), that fallback and its `ValidationException` stay in the connector, applied as an explicit `region()` override on top of `AwsClientSupport.configureClient`. bedrock, s3 and sagemaker have no such fallback and call `AwsClientSupport.createClient` directly.

### Left unmigrated

Nothing. All 7 connectors' hand-rolled client-construction sites now go through `AwsClientSupport`, including aws-sqs's `AmazonSQSClientSupplier`, which was reshaped from `sqsClient(credentialsProvider, region[, endpoint])` to `sqsClient(request, region)` (mirroring `DynamoDbClientSupplier`) to make that possible. The only AWS client construction anywhere in `connectors/aws/**` not going through `AwsClientSupport` is `SnsClientSupplier.messageManager(String region)`, which still uses AWS SDK v1 - there is no v2 equivalent (tracked separately in #7974).

### Behavior changes

- **Endpoint override newly honored for bedrock, s3 and sagemaker.** `configuration.endpoint` is a real (hidden) property on these connectors' element templates that the previous hand-rolled builders silently ignored. It's now applied like every other AWS connector. No UI-driven caller sets it today since it's hidden from the Modeler panel, so this only changes behavior for a hand-edited or hybrid-template process definition that already sets it. s3 has no `forcePathStyle` option, so a custom endpoint there uses virtual-host-style addressing and won't work against LocalStack/MinIO out of the box.
- **Endpoint override newly honored for sqs's inbound connector**, which never called the old 3-arg overload at all.
- **A blank (empty-string, non-null) endpoint is now a no-op instead of a construction-time failure**, for lambda, sns, eventbridge and sqs. The previous inline builders fed any non-null endpoint straight to `endpointOverride(URI.create(...))`, which throws for `""`. `AwsClientSupport` treats a blank endpoint the same as an absent one - the same fix already shipped for aws-dynamodb. No test in any of these connectors set `configuration.endpoint`, so this edge case was unreachable before and stays behavior-safe now.
- **A missing/blank region on bedrock, s3 and sagemaker now falls through to the AWS SDK's default region-provider chain instead of failing fast** (previously `NullPointerException`/`IllegalArgumentException` at client-construction time). This is inherent to `AwsClientSupport` itself, which has no region validation, and is already the behavior of every other connector already on this helper (aws-dynamodb, aws-textract, aws-comprehend, aws-bedrock-knowledgebase) - not something introduced by this PR. Calling it out because it's a real, if narrow, behavior change: a missing region is only reachable via a hand-edited or FEEL-driven process definition, since the element template enforces "not empty" through UI metadata, not a server-side Jakarta constraint. A fix belongs in `AwsClientSupport` itself as a follow-up, not in the three connectors that happen to be touched here.

## Tests

Added a supplier-level test per connector (mirroring the existing aws-dynamodb pattern) that builds a real client through the new code path with region-only, endpoint-set and blank-endpoint configurations, plus a shared `AwsClientSupportTest` in aws-base that pins the exact calls `configureClient` makes on the client builder (credentials provider, region, endpoint). Also tightened a pre-existing lambda test that was asserting region resolution with an untyped matcher instead of the actual expected value. `EventBridgeFunctionTest` asserts the request handed to the supplier carries the secret-substituted credentials; wiring those onto a client builder is covered once, in `AwsClientSupportTest`.

Ran the full test suite for all 7 touched connectors plus aws-base; all green, `dependency:analyze` clean, `spotless:check` clean, element templates unchanged. Also compiled the `connectors-e2e-test-aws-*` modules against these changes to confirm nothing outside the touched connectors depends on the reshaped `AmazonSQSClientSupplier` signature.

## Related issues

closes #7083

## Checklist

- [ ] Backport labels are added if these code changes should be backported. — not applicable: no released branch is affected, this only touches `main`.
- [x] Tests/Integration tests for the changes have been added if applicable.
- [ ] If the change requires a documentation update, it has been added to the appropriate section in the documentation. — none needed: no documented connector behavior changed; the endpoint-override note above is a capability gain already reflected in the existing element templates.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: connectors/aws/aws-lambda/src/test/java/io/camunda/connector/awslambda/LambdaConnectorFunctionTest.java [hunk 1]
```diff
@@ -57,6 +61,16 @@
...
     assertThat(result.getPayload()).isEqualTo(ACTUAL_PAYLOAD);
+    // Then the region actually passed to the client supplier must be the one AwsUtils computes
+    // from the bound request (configuration.region, falling back to the deprecated
+    // awsFunction.region) - several success cases omit configuration.region entirely and rely on
+    // that fallback, so this pins LambdaConnectorFunction's own region-resolution call rather than
+    // letting an untyped any() swallow a wrong-source or swapped-argument regression.
+    var request = context.bindVariables(AwsLambdaRequest.class);
+    var expectedRegion =
+        AwsUtils.extractRegionOrDefault(
+            request.getConfiguration(), request.getAwsFunction().getRegion());
```
[the remark this answers is not in the document]
Comment (ztefanie) [reply connectors/aws/aws-lambda/src/test/java/io/camunda/connector/awslambda/LambdaConnectorFunctionTest.java]: @chillleader Can you fix this, please?

**Review by ztefanie — APPROVED** [review]
Looks good and also cleaner 🚀 

Will older templates work with newer runtime or is this breaking?

(optional) AI added a lot of (for my feeling) unnecessary multiline comments, that no one will maintain. I would prefer to remove them, but up to you.

**Comment by chillleader** [comment]
@ztefanie yes, should be fully backward / forward compatible, ensured by tests I added in the beginning of this epic, and no element template changes
